File: backend/src/app/models/enrollment.ts
/**
 * @file enrollment.ts
 * @description
 * Enrollment related models
 *
 * @module models
 * @version 1.0.0
 * @auth Thomas
 */
type Enrollment = {
  id: number;
  courseId: number;
  userId: number;
  createdAt: Date;
  updatedAt: Date;
};

export type { Enrollment };

File: SchoolSpace-Server/src/app/models/core/Enrollment.cs
namespace backend.app.models.core
{
    public class Enrollment : ITimestamped
    {
        public int UserId { get; set; }
        public User User { get; set; } = null!;
        public int CourseId { get; set; }
        public Course Course { get; set; } = null!;
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
    }
}
